src/data_model/transformations.jl

```diff
@@ -472,3 +472,917 @@ function apply_voltage_angle_difference_bounds!(eng::Dict{String,<:Any}, vad::Re
+    end
+
+    return data_eng
+end
+
+
+"Join lines which are connected in series, and of which the intermediate bus is ungrounded and does not connect to any other components."
+function join_lines!(data_eng::Dict{String,Any})
+    @assert data_eng["data_model"]==ENGINEERING
+
+    # a bus is eligible for reduction if it only appears in exactly two lines
+    buses_all = collect(keys(data_eng["bus"]))
+
+    # exclude buses that appear in components other than lines
+    comp_types_exclude = setdiff(pmd_eng_asset_types, ["line"])
+    buses_exclude = get_defined_buses(data_eng, comp_types=comp_types_exclude)
+    # exclude buses that are grounded
+    grounded_buses = [bus_id for (bus_id,bus) in data_eng["bus"] if !isempty(bus["grounded"])]
+    buses_exclude = union(buses_exclude, grounded_buses)
+
+    # per bus, list all inbound or outbound lines
+    bus_lines = Dict(bus=>[] for bus in buses_all)
+    for (id,line) in data_eng["line"]
+        push!(bus_lines[line["f_bus"]], id)
+        push!(bus_lines[line["t_bus"]], id)
+    end
+
+    # exclude all buses that do not have exactly two lines connected to it
+    buses_exclude = union(buses_exclude, [bus for (bus,lines) in bus_lines if length(lines)!=2])
+
+    # now loop over remaining buses
+    candidates = setdiff(buses_all, buses_exclude)
+    for bus in candidates
+        line1_id, line2_id = bus_lines[bus]
+        line1 = data_eng["line"][line1_id]
+        line2 = data_eng["line"][line2_id]
+
+        # reverse lines if needed to get the order
+        # (x)--fr-line1-to--(bus)--to-line2-fr--(x)
+        if line1["f_bus"]==bus
+            _line_reverse!(line1)
+        end
+        if line2["f_bus"]==bus
+            _line_reverse!(line2)
+        end
+
+        reducable = true
+        reducable = reducable && line1["linecode"]==line2["linecode"]
+        reducable = reducable && all(line1["t_connections"].==line2["t_connections"])
+        if reducable
+
+            line1["length"] += line2["length"]
+            line1["t_bus"] = line2["f_bus"]
+            line1["t_connections"] = line2["f_connections"]
+
+            delete!(data_eng["line"], line2_id)
+            delete!(data_eng["bus"], bus)
+            for x in candidates
+                if line2_id in bus_lines[x]
+                    bus_lines[x] = [setdiff(bus_lines[x], [line2_id])..., line1_id]
+                end
+            end
+        end
+    end
+
+    return data_eng
+end
+
+
+"""
+Adds absolute (i.e. indivdially, not between a pair of terminals) voltage bounds through the 'vm_lb' and 'vm_ub' property.
+Bounds are specified in per unit, and automatically converted to SI units by calculating the voltage base. 
+If you change data_eng["settings"]["vbases_default"], the data model transformation will however produce inconsistent bounds in per unit.
+Neutral terminals are automatically detected, and set to [0,phase_ub_pu*vbase]. 
+"""
```

Comment:
For exported functions, we should follow the new docstring pattern where the function is given (indented) first, see, e.g. transform_data_model function as an example.